src/main/java/io/confluent/connect/elasticsearch/ElasticsearchClient.java

```diff
@@ -108,13 +111,15 @@
   private final RestHighLevelClient client;
   private final ExecutorService bulkExecutorService;
   private final Time clock;
+  private final Lock lock = new ReentrantLock();
+  private final Condition inFlightRequestsUpdated = lock.newCondition();
 
   // Visible for testing
   public ElasticsearchClient(
-          ElasticsearchSinkConnectorConfig config,
-          ErrantRecordReporter reporter
+      ElasticsearchSinkConnectorConfig config,
+      ErrantRecordReporter reporter
```

Comment:
should we instead pass OffsetTracker as argument to constructor and just relay it? This way testcases can even pass mocks that arent one of the below two OffsetTracker implmentations.

Reply:
Good point, we don't need this additional constructor just for testing then.